torch/_guards.py

```diff
@@ -175,3 +176,31 @@ def set_export_info(self, guard_type, guarded_class, code_list, obj_weakref):
             None,
         ), "Guarded object must be identical, or None"
         self.obj_weakref = obj_weakref
+
+
+class GuardsContext:
+    def __init__(self):
+        self.dynamo_guards: Set[Guard] = set()
+
+
+_CURRENT_TRACING_CONTEXT = None
+
+
+class TracingContext:
+    @staticmethod
+    def get() -> Optional["TracingContext"]:
+        return _CURRENT_TRACING_CONTEXT
+
+    def __init__(self):
+        self.guards_context = GuardsContext()
+
+
+@contextmanager
```

Comment:
Is this second func necessary? I would expect the `TracingContext` to be a context manager. Then whenever you call `with <some_context>:` it can automatically do the swapping with the global var and itself.